python/lsst/pipe/supertask/super_task.py

```diff
@@ -223,103 +119,69 @@ def completed(self, completed):
         self._completed = completed
 
-    def pre_run(self, *args, **kwargs):
-        """
-        Prerun method, which given the decorator @wraprun will run before run()
-        This method can be used to prepare the data or logs before running the task
-        """
-        pass
-
-    def post_run(self, *args, **kwargs):
-        """
-        Postrun method, this hook method can be used to define the completion of a task and to
-        gather information about the run()
-        """
-        # By default, if used, this methods set the completion to be true.
-        if args is None and kwargs in None:
-            self.completed = True
-
     def run(self, *args, **kwargs):
         """
-        This function is the one that actually operates on teh data (exposed by execute) and usually
-        returning a Struct with the collected results
+        Run task algorithm on in-memory data.
+
+        This function is the one that actually operates on the data (exposed by execute) and
+        usually returning a Struct with the collected results. This method will be overiden by
+        every subclass. It operates on in-memory data structures (or data proxies) and cannot
+        access any external data such as data butler or databases. All interaction with external
+        data happens in `execute()` method.
         """
         pass
 
-    @property
-    def parser(self):
-        """
-        Hook to handle the parser, it might be deprecated soon
-        :return: the parser used to call this task
+    def execute(self, dataRef):
         """
-        return self._parser
+        Retrieve data and run task algorithm on it.
 
-    @parser.setter
-    def parser(self, parser):
-        """
-        parser setter
-        """
-        self._parser = parser
+        This method is responsible for handling access to external data. It retrieves (probably
+        lazily) data defined by `dataRef`, calls `run()` method on that data, and stores
+        resulting data in butler. This method needs to be overriden by subclasses.
 
-    def execute(self, dataRef):
-        """
-        execute method: This is the method that differentiate Task from SuperTask, this will
-        probably change in the future, but as for now this method takes a dataRef (it might be a
-        list of dataRef) and then gets the images, tables or anything needed for the run method to
-        operate. The execute method should NOT process the object but it should ALWAYS call the
-        run method with the necessary inputs.
-
-        :param dataRef: Butler dataRef (or list of dataRef eventually) which will be inspect to
-        get necessary inputs for run. All SuperTask need to define a execute and call run from
-        inside.
+        :param dataRef: Butler dataRef (or list of dataRef eventually) which will be inspected to
+        get necessary inputs for run.
         :return: the results from run()
         """
         pass
 
-    def get_conf_list(self):
```

Comment:
This does not look like something useful to the clients (activator). I wanted to make it a private method, but then decided it's easier to inline it in `print_config()` (that method is also questionable). If activator needs something like this then we can discuss it, but you already have access to `task.config` so it's easy to do whatever you want in activator.